packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -33,32 +38,73 @@ export class Wallet extends EventEmitter<ObjectiveProposed> {
    */
   public static async create(
     engine: Engine,
+    chainService: ChainServiceInterface,
     messageServiceFactory: MessageServiceFactory,
     retryOptions: Partial<RetryOptions> = DEFAULTS
   ): Promise<Wallet> {
-    return new Wallet(messageServiceFactory, engine, {...DEFAULTS, ...retryOptions});
+    await chainService.checkChainId(engine.engineConfig.networkConfiguration.chainNetworkID);
+    const wallet = new Wallet(messageServiceFactory, engine, chainService, {
+      ...DEFAULTS,
+      ...retryOptions,
+    });
+    await wallet.registerExistingChannelsWithChainService();
+    return wallet;
   }
 
   private _messageService: MessageServiceInterface;
+  private _destroyed = false;
```

Comment:
Is this used?

Reply:
No I don't think so. I did a lot of rebasing re-ordering of PRs so I might of included it by accident.

Reply:
Done in https://github.com/statechannels/statechannels/pull/3555/commits/deae1e90e3fde6cc5cae1fcdf8a9ae3c19542b46